### PR TITLE
[DX-783] Add the session management and audit log settings for the EDP.

### PR DESCRIPTION
**Pull request by slega**
### For internal users - Please add a Jira DX PR ticket to the subject!
https://tyktech.atlassian.net/browse/DX-783

---

### Preview Link
https://deploy-preview-3502--tyk-docs.netlify.app/docs/nightly/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration/

### Description
Add documentations for the following config options:
- PORTAL_THEMING_DISABLE_UPLOAD
- PORTAL_MAX_UPLOAD_SIZE
- PORTAL_DCR_LOG_ENABLED
- PORTAL_AUDIT_LOG_ENABLE
- PORTAL_AUDIT_LOG_PATH
- PORTAL_SESSION_NAME
- PORTAL_SESSION_SECURE
- PORTAL_SESSION_HTTPONLY
- PORTAL_SESSION_KEY
- PORTAL_SESSION_LIFETIME
- PORTAL_TLS_ENABLE
- PORTAL_TLS_CERTIFICATES
<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

### CODE REVIEW

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 1]
```diff
@@ -103,6 +113,62 @@ The default value for this variable is `./themes`, so it's important to redefine
...
+**Description**: Sets the `HttpOnly` attribute on the portal's cookie which controls is the portal's cookie is only accessible by the server and not by javascript on the client side.
+This is a security measure to prevent XSS attacks. We recommend setting it to `true` in production environments. The default value is `true`.
+
+#### PORTAL_SESSION_KEY
+**Config file:** Session.Key <br/>
+**Type:** `string` <br/>
+**Description**: The cookie encryption key. The default value is a random 32-bytes string.
+
+#### PORTAL_SESSION_LIFETIME
+*Config file:** Session.LifeTime <br/>
```
Comment (JRWu) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]: Minor formatting suggestion- I believe this is a typo, should be two stars  i.e. **Config file:** in order to render properly

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 2]
```diff
@@ -227,6 +293,28 @@ If the bucket uses a policy to set permissions, you should leave the ACL value e
...
+### TLS configuration
+This section provides a reference for the TLS configuration settings to enable connection to the portal's UI over https.
+
+#### PORTAL_TLS_ENABLE
+**Config file:** TLSConfig.Enable <br/>
+**Type:** `boolean` <br/>
+**Description**: Enables or disables connection over https. When TLS is enabled, the portal will expect a TLS certificate to be provided via PORTAL_TLS_CERTIFICATES.
+When TLS is enabled and no certificates provided, the portal won't start. The default value is `false`.
+
+#### PORTAL_TLS_CERTIFICATES
```
Comment (JRWu) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md:305]: Should we also include what format of cert file the certificate must be in as well just so prospects don't get confused? I know the dashboard has a section in the documentation ([linked here](https://tyk.io/docs/basic-config-and-security/security/tls-and-ssl/#using-tyk-certificate-storage)) that illustrates what formatting must be applied to the certificates.

**Review by JRWu — CHANGES_REQUESTED** [review]
Looks great, just requesting a couple minor changes

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 3]
```diff
@@ -103,6 +113,62 @@ The default value for this variable is `./themes`, so it's important to redefine
 
 **Description**: Use this setting to specify which OAS documentation renderer to use to render Open API Specification. Not required. If it is not specified, the default value is `stoplight`.
 
+#### PORTAL_DCR_LOG_ENABLED
+**Config file:** DCRLogEnabled <br/>
+**Type:** `boolean` <br/>
+**Description**: When enabled, the portal will print raw responses from OAuth2.0 Identity Provider for the DCR flow.
+Raw responses from the Identity Providers may contain sensitive information therefore we recommend enabling this option only for debugging purposes. Available options are:
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
Raw responses from the Identity Providers may contain sensitive information, therefore we recommend enabling this option only for debugging purposes. Available options are:
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 4]
```diff
@@ -103,6 +113,62 @@ The default value for this variable is `./themes`, so it's important to redefine
...
+- `true` for enabling the detailed logs;
+- `false` for disabling the detailed logs.
+The default value is `false`.
+
+### Audit log settings
+This section explains how to configure the audit log in the portal. When the audit log enabled, each admins' action will leave a trace in the portal.log file located at in the directory specified by the `PORTAL_AUDIT_LOG_ENABLE` setting.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
This section explains how to configure the audit log in the portal. When the audit log is enabled, each admins' action will leave a trace in the *portal.log* file located at in the directory specified by the `PORTAL_AUDIT_LOG_ENABLE` setting.
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 5]
```diff
@@ -103,6 +113,62 @@ The default value for this variable is `./themes`, so it's important to redefine
...
+**Type:** `boolean` <br/>
+**Description**: Enables the audit log capability. The default value is `false`.
+
+#### PORTAL_AUDIT_LOG_PATH
+**Config file:** AuditLog.Path <br/>
+**Type:** `string` <br/>
+**Description**: Path to a directory with the audit log file. When audit log is enabled, the portal will create a file called `portal.log` in that directory. All admin actions will be reflected in that file.
+
+### Session management
+This section explains how to configure the session management settings in the portal. Using the below setting, you can configure:
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
This section explains how to configure session management for the portal. Using the settings below, you can configure:
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 6]
```diff
@@ -103,6 +113,62 @@ The default value for this variable is `./themes`, so it's important to redefine
...
+- The name of the portal's session cookie;
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
- Name of the portal's session cookie.
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 7]
```diff
@@ -103,6 +113,62 @@ The default value for this variable is `./themes`, so it's important to redefine
...
+- Various aspects of cookie security such as should it be sent only over a TLS-encrypted connection and is accessible by Javascript API on the client-side;
+- The cookie encryption key;
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
- Cookie encryption key.
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 8]
```diff
@@ -103,6 +113,62 @@ The default value for this variable is `./themes`, so it's important to redefine
...
+
+#### PORTAL_AUDIT_LOG_PATH
+**Config file:** AuditLog.Path <br/>
+**Type:** `string` <br/>
+**Description**: Path to a directory with the audit log file. When audit log is enabled, the portal will create a file called `portal.log` in that directory. All admin actions will be reflected in that file.
+
+### Session management
+This section explains how to configure the session management settings in the portal. Using the below setting, you can configure:
+- The name of the portal's session cookie;
+- Various aspects of cookie security such as should it be sent only over a TLS-encrypted connection and is accessible by Javascript API on the client-side;
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
- Various aspects of cookie security, including: should it be sent using an TLS-encrypted connection and is it accessible by Javascript API on the client-side?
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 9]
```diff
@@ -103,6 +113,62 @@ The default value for this variable is `./themes`, so it's important to redefine
...
+#### PORTAL_SESSION_SECURE
+**Config file:** Session.Secure <br/>
+**Type:** `boolean` <br/>
+**Description**: Sets the `Secure` attribute on the portal's cookie which controls if the portal's cookie is sent only over HTTPS connections. We recommend to set it to `true` when TLS for the portal is enabled.
+When TLS is not enabled for the portal, setting this setting to `true` will prevent the portal from sending this cookie to the backend rendering the portal non-operational. We recommend enabling TLS and setting this attribute to `true` for all production environments. Default value is `false`.
+
+#### PORTAL_SESSION_HTTPONLY
+**Config file:** Session.HttpOnly <br/>
+**Type:** `boolean` <br/>
+**Description**: Sets the `HttpOnly` attribute on the portal's cookie which controls is the portal's cookie is only accessible by the server and not by javascript on the client side.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
**Description**: Sets the `HttpOnly` attribute on the portal's cookie which controls if the cookie is only accessible at the server and not by Javascript on the client side.
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 10]
```diff
@@ -103,6 +113,62 @@ The default value for this variable is `./themes`, so it's important to redefine
...
+
+#### PORTAL_SESSION_KEY
+**Config file:** Session.Key <br/>
+**Type:** `string` <br/>
+**Description**: The cookie encryption key. The default value is a random 32-bytes string.
+
+#### PORTAL_SESSION_LIFETIME
+*Config file:** Session.LifeTime <br/>
+**Type:** `int` <br/>
+**Description**: The lifetime of the portal's cookie in seconds. the default value is 3600 seconds.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
**Description**: The lifetime of the portal's cookie in seconds. The default value is 3600 seconds.
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 11]
```diff
@@ -125,7 +191,7 @@ The default value for this variable is `./themes`, so it's important to redefine
 #### PORTAL_TLS_ENABLE
 **Config file:** TLSConfig.Enable <br/>
 **Type:** `boolean` <br/>
-**Description**: Enables TLS.
+**Description**: Enables TLS. The default value is false.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
**Description**: Enables TLS. The default value is `false`.
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 12]
```diff
@@ -227,6 +293,28 @@ If the bucket uses a policy to set permissions, you should leave the ACL value e
 This is particularly useful if the bucket is private and you need to access the object directly, such as when displaying an image on a web page.
 This option is only required for the `s3` storage type and will be ignored for the `fs` and `db` storage types.
 
+### TLS configuration
+This section provides a reference for the TLS configuration settings to enable connection to the portal's UI over https.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
This section explains the TLS configuration settings to enable connection to the portal's UI over HTTPS.
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 13]
```diff
@@ -227,6 +293,28 @@ If the bucket uses a policy to set permissions, you should leave the ACL value e
...
+
+#### PORTAL_TLS_ENABLE
+**Config file:** TLSConfig.Enable <br/>
+**Type:** `boolean` <br/>
+**Description**: Enables or disables connection over https. When TLS is enabled, the portal will expect a TLS certificate to be provided via PORTAL_TLS_CERTIFICATES.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
**Description**: Enables or disables connection over https. When TLS is enabled, the portal will expect a TLS certificate to be provided via *PORTAL_TLS_CERTIFICATES*.
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 14]
```diff
@@ -227,6 +293,28 @@ If the bucket uses a policy to set permissions, you should leave the ACL value e
...
+When TLS is enabled and no certificates provided, the portal won't start. The default value is `false`.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
When TLS is enabled and no certificates are provided, the portal won't start. The default value is `false`.
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 15]
```diff
@@ -227,6 +293,28 @@ If the bucket uses a policy to set permissions, you should leave the ACL value e
...
+
+#### PORTAL_TLS_CERTIFICATES
+**Config file:** TLSConfig.Certificates <br/>
+**Type:** `string` <br/>
+**Description**: A json-formatted string that provides hostname and paths to a cert and a key file. To work correctly, the portal need the following properties in the config string:
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
**Description**: A JSON formatted string that provides the hostname , in addition to the paths to a TLS certificate and key file:
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 16]
```diff
@@ -227,6 +293,28 @@ If the bucket uses a policy to set permissions, you should leave the ACL value e
...
+- `Name`: a hostname of the portal, should match with the hostname of the certificate file.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]: Do we need specify Common Name(CN) / Subject Alternative Name (SAN) in certificate for matching hostname?

- `Name`: The hostname of the portal. This should match with the Common Name (CN) or Subject Alternative Name (SAN) specified in the certificate file.

```suggestion
- `Name`: The hostname of the portal. This should match with the hostname of the certificate file.
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 17]
```diff
@@ -227,6 +293,28 @@ If the bucket uses a policy to set permissions, you should leave the ACL value e
...
+- `CertFile`: a path to a TLS certificate file for the specified hostname.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
- `CertFile`: The path to a TLS certificate file for the specified hostname.
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md [hunk 18]
```diff
@@ -227,6 +293,28 @@ If the bucket uses a policy to set permissions, you should leave the ACL value e
...
+- `KeyFile`: a path to a TlS key file for the specified hostname.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md]:
```suggestion
- `KeyFile`: The path to a TLS key file for the specified hostname.
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Hi @slega,

Many thanks for submitting the PR with updated config settings.

PR looks good to me with some minor suggestions. Once suggestions have been committed and peer review comment addressed then we can merge and release.

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md (hunk 2 above)
Comment (slega) [reply tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration.md:305]: @JRWu the documentation you've provided is for Mutual TLS for upstream / downstream. The documentation for the dashboard doesn't specify the format: https://tyk.io/docs/basic-config-and-security/security/tls-and-ssl/. But I've added the format (.crt) for the certificates.

**Review by JRWu — APPROVED** [review]
LGTM!

**Review by dcs3spp — APPROVED** [review]
Hi Sergei, PR looks good to me

**Pull request merged**